### PR TITLE
Structure the templates directory

### PR DESCRIPTION
**Pull request by stephanegigandet**
This starts to put a little bit of structure in the templates directory, as suggested by @CharlesNepote in #5261

- As we now have emails templates in templates/emails/ , I moved all web pages related templates to templates/web/
- The not intuitively display_new template is now renamed to web/site_layout.html (that's the template that has all the menus, side column and footer)
- in templates/web there are a pages/ and a common/ directory:
- templates/web/common/ is for things that are common to several pages (e.g. the donate_banner that is included by site_layout)
- templates/pages contains the different types of pages, in sub directories:
- I created the dir for only one page, as an example: templates/pages/product contains all the product page related templates
- templates/pages/product/product_page.tt.html : template ending in _page is the main page template
- templates/pages/product/edit_history.tt.html etc. (not suffixed by _page) are components of the page.

If everything looks good, we can continue to move all the other templates to their respective pages, and then continue with Charles' other suggestions like adding a comment at the top of each template to indicate which part of the page is generated by which template.

### CODE REVIEW

File: templates/web/pages/product/forest_footprint_details.tt.html [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<p>
```
Comment (CharlesNepote) [inline templates/web/pages/product/forest_footprint_details.tt.html:1]:
```suggestion
<!-- [% template.name %] -->
<p id="forest_footprint_calculation_details">
```

[% template.name %] should print the template name.

File: templates/web/pages/product/forest_footprint_details.tt.html [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	[% ELSE %]
+		<p>Les informations sur l'emballage sont manquantes, son impact n'a pas pu être pris en compte.</p>
+	[% END %]
+	
+	<h3>Score final</h3>
+
+	<p><strong>[% lang('ecoscore_score') %][% sep %]: [% round(score) %]</strong></p>
+	<p><strong>[% lang('ecoscore_grade') %][% sep %]: [% grade FILTER ucfirst %]</strong></p>
+
+</div>
```
Comment (CharlesNepote) [inline templates/web/pages/product/forest_footprint_details.tt.html:35]:
```suggestion
</div>
<!-- / [% template.name %] -->
```

**Review by CharlesNepote — APPROVED** [review]
I would have used a different name for templates related to the whole page and templates related to parts of pages (the later could have an "inc" prefix). "inc" let you know it's a piece/component and not the whole page. But maybe _page is sufficient.

But it's already a good improvement.

I made 2 suggestions, feel free to integrate it or not.

File: templates/web/pages/product/forest_footprint_details.tt.html (hunk 1 above)
Comment (stephanegigandet) [reply templates/web/pages/product/forest_footprint_details.tt.html:1]: Perfect, I didn't know [% template.name %]

**Comment by stephanegigandet** [comment]
> I would have used a different name for templates related to the whole page and templates related to parts of pages (the later could have an "inc" prefix). "inc" let you know it's a piece/component and not the whole page.

I would prefer to create a specific includes directory instead of using file name prefixes. Something like templates/web/pages/products/includes/ I'll try to see how it looks.

It would also work well for the templates/web/common directory, to have the site_layout.tt.html in common/ and things like the donate_banner.tt.html in common/includes/

**Comment by roshnaeem** [comment]
Thank you @stephanegigandet for working on this. I just have a small question. How are we gonna deal with `.tt.js` files. Are we gonna make new separate `js` directory/directories for them or these files will be in the same folder as their corresponding `tt.html` files?

**Comment by stephanegigandet** [comment]
> How are we gonna deal with `.tt.js` files. Are we gonna make new separate `js` directory/directories for them or these files will be in the same folder as their corresponding `tt.html` files?

@roshnaeem: Maybe we can just leave them in the same folder as the corresponding tt.html file for now.

**Comment by stephanegigandet** [comment]
Merging as I have other bug fixes that depend on this. We can always adjust the directory structure later if needed.

**Pull request merged**